python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_freshness.py

```diff
@@ -0,0 +1,104 @@
+from dagster._core.definitions.events import AssetObservation
+from dagster._core.events import AssetObservationData
+from dagster._core.events.log import EventLogEntry
+from dagster._core.workspace.context import WorkspaceRequestContext
+from dagster_graphql.test.utils import (
+    execute_dagster_graphql,
+)
+from dagster import DagsterEvent, DagsterEventType
+import time
+from dagster_graphql_tests.graphql.graphql_context_test_suite import (
+    ExecutingGraphQLContextTestMatrix,
+)
+
+GET_FRESHNESS_INFO = """
+    query AssetNodeQuery($assetKeys: [AssetKeyInput!]!) {
+        assetNodes(assetKeys: $assetKeys) {
+            id
+            freshnessInfo {
+                currentMinutesLate
+                latestMaterializationMinutesLate
+            }
+            freshnessPolicy {
+                cronSchedule
+                maximumLagMinutes
+            }
+        }
+    }
+"""
+
+
+class TestAssetFreshness(ExecutingGraphQLContextTestMatrix):
```

Comment:
Somewhat philosophical question not fully related to this PR - do we need the full test matrix for functionality like this?

Reply:
Good point, it's easy to cargo cult that. The other way that gql tests feel bloated is the fact that we snapshot everything in the repo. Makes it a pain to add anything